src/runtime_src/driver/xclng/tools/xbutil/xbutil.cpp

```diff
@@ -1479,3 +1493,174 @@ int xcldev::xclP2p(int argc, char *argv[])
 
     return ret;
 }
+
+/*
+ * m2mtest
+ */
+static void m2m_free_unmap_bo(xclDeviceHandle handle, unsigned boh, 
+auto * &boptr, size_t boSize)
+{
+    if(boh != NULLBO)
+        xclFreeBO(handle, boh);
```

Comment:
Logically, you want to unmap the BO, then free it.

Reply:
Done